Project-Aurora/Project-Aurora/Devices/Omen/OmenDevices.cs

```diff
@@ -0,0 +1,213 @@
+    {
+        bool kbConnected = false;
+        bool peripheralConnected = false;
+
+        List<IOmenDevice> devices;
+
+        private bool isInitialized = false;
+        private readonly string devicename = "OMEN";
+
+        private readonly System.Diagnostics.Stopwatch watch = new System.Diagnostics.Stopwatch();
+        private long lastUpdateTime = 0;
+
+        public string GetDeviceDetails()
+        {
+            return devicename;
+        }
+
+        public string GetDeviceName()
+        {
+            if (isInitialized)
+            {
+                string result = devicename + ":";
+                foreach (var dev in devices)
+                {
+                    if (dev.GetDeviceName() != string.Empty)
+                    {
+                        result += (" " + dev.GetDeviceName() + ";");
+                    }
+                }
+
+                return result;
+            }
+            else
+            {
+                return devicename + ": Not initialized";
+            }
+        }
+
+        public string GetDeviceUpdatePerformance()
+        {
+            return (isInitialized ? lastUpdateTime + " ms" : "");
+        }
+
+        public VariableRegistry GetRegisteredVariables()
+        {
+            return new VariableRegistry();
+        }
+
+        public bool Initialize()
+        {
+
+            lock (this)
+            {
+                if (!isInitialized)
+                {
+                    devices = new List<IOmenDevice>();
+                    IOmenDevice dev;
+                    if ((dev = OmenKeyboard.GetOmenKeyboard()) != null)
+                    {
+                        devices.Add(dev);
+                        kbConnected = true;
+                    }
+                    else if ((dev = OmenFourZoneLighting.GetFourZoneLighting()) != null)
+                    {
+                        devices.Add(dev);
+                        kbConnected = true;
+                    }
+
+                    if ((dev = OmenMouse.GetOmenMouse()) != null)
+                    {
+                        devices.Add(dev);
+                        peripheralConnected = true;
+                    }
+
+                    if ((dev = OmenMousePad.GetOmenMousePad()) != null)
+                    {
+                        devices.Add(dev);
+                        peripheralConnected = true;
+                    }
+
+                    if ((dev = OmenChassis.GetOmenChassis()) != null)
+                    {
+                        devices.Add(OmenChassis.GetOmenChassis());
+                        peripheralConnected = true;
+                    }
+
+                    if ((dev = OmenSpeaker.GetOmenSpeaker()) != null)
+                    {
+                        devices.Add(dev);
+                        peripheralConnected = true;
+                    }
+
+                    Global.kbLayout.KeyboardLayoutUpdated += DeviceChangedHandler;
+
+                    isInitialized = true;
```

Comment:
Initialize always returns true, even if the user has no omen devices.

Reply:
Fixed it